### PR TITLE
cthalmann/add-tests

### PR DESCRIPTION
**Pull request by cthalmann**
Added tests for cookbook, passes rake.

### CODE REVIEW

**Comment by fahlmant** [comment]
:+1:

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  platform: 'centos',
+  version: '6.8',
+  log_level: :fatal
+}.freeze
+
+DEBIAN_8_OPTS = {
+  platform: 'debian',
+  version: '8.4',
+  log_level: :fatal
+}.freeze
```
Comment (armiller) [inline spec/spec_helper.rb]: This cookbook doesn't support debian so this can be removed.

File: spec/client_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    context "on #{pltfrm[:platform]} #{pltfrm[:version]}" do
+      cached(:chef_run) do
+        ChefSpec::SoloRunner.new(pltfrm).converge(described_recipe)
+      end
+      before do
+        stub_data_bag_item('users', 'rdiff-backup-client')
+      end
+      it do
+        expect { chef_run }.to_not raise_error
+      end
```
Comment (armiller) [inline spec/client_spec.rb]: Remove these raise_error tests.

File: spec/client_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        )
+      end
+      it do
+        expect(chef_run).to reload_ohai('reload_passwd').with(plugin: 'etc')
+      end
+      it do
+        expect(chef_run).to install_package('rdiff-backup')
+      end
+      it do
+        %w(yum yum-epel ssh-keys).each do |r|
```
Comment (armiller) [inline spec/client_spec.rb]: This loop needs to be outside the it block.

Comment (armiller) [inline spec/client_spec.rb]:
```ruby
%w(foo bar).each do |r|
  it do
    expect(chef_run).to include_recipe(r)
  end
end

File: spec/client_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+require_relative 'spec_helper'
+
+describe 'rdiff-backup::client' do
+  [CENTOS_6_OPTS,
+   CENTOS_7_OPTS].each do |pltfrm|
```
Comment (armiller) [inline spec/client_spec.rb]: This is short enough to be one line.

File: spec/default_spec.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+require_relative 'spec_helper'
+
+describe 'rdiff-backup::default' do
+  [CENTOS_6_OPTS,
+   CENTOS_7_OPTS].each do |pltfrm|
```
Comment (armiller) [inline spec/default_spec.rb]: same here

**Comment by fahlmant** [comment]
:+1:

File: spec/client_spec.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  [CENTOS_6_OPTS, CENTOS_7_OPTS].each do |pltfrm|
+    context "on #{pltfrm[:platform]} #{pltfrm[:version]}" do
+      cached(:chef_run) do
+        ChefSpec::SoloRunner.new(pltfrm).converge(described_recipe)
+      end
+      before do
+        stub_data_bag_item('users', 'rdiff-backup-client')
+      end
+      it do
+        expect(chef_run).to create_user('rdiff-backup-client').with(
```
Comment (DesertBeagle) [inline spec/client_spec.rb]: Needs to check for line 30 in the recipe:
`supports manage_home: true`

File: spec/client_spec.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+          comment: 'User for rdiff-backup client backups',
+          shell: '/bin/bash',
+          action: [:nothing]
+        )
+      end
+      it do
+        expect(chef_run).to reload_ohai('reload_passwd').with(plugin: 'etc')
```
Comment (DesertBeagle) [inline spec/client_spec.rb]: `action: nothing`
as welll

File: spec/client_spec.rb [hunk 8]
```diff
@@ -0,0 +1,44 @@
+require_relative 'spec_helper'
+
+describe 'rdiff-backup::client' do
+  [CENTOS_6_OPTS, CENTOS_7_OPTS].each do |pltfrm|
+    context "on #{pltfrm[:platform]} #{pltfrm[:version]}" do
+      cached(:chef_run) do
+        ChefSpec::SoloRunner.new(pltfrm).converge(described_recipe)
```
Comment (DesertBeagle) [inline spec/client_spec.rb]: Needs to test line 45
`node.override['ssh_keys'][client_user] = ['rdiff-backup-client']`
example: https://github.com/sethvargo/chefspec/blob/master/examples/attributes/spec/default_spec.rb

Comment (DesertBeagle) [inline spec/client_spec.rb]: There's no good way to test this. I would just ignore it and if anyone has a problem, ask them for a good way to do it.

Comment (armiller) [inline spec/client_spec.rb]: You can test the node attribute.
```ruby
expect(chef.node['ssh_keys']['foo']).to eq(['rdiff-backup-client'])
```

**Comment by DesertBeagle** [comment]
Could you rebase this?

**Comment by DesertBeagle** [comment]
:+1:

**Comment by fahlmant** [comment]
:+1:

**Pull request merged**